### PR TITLE
Handle patterns of the form H[a_Type,a_Type,...]

### PR DESCRIPTION
**Pull request by mmatera**
This PR handles another corner case of patterns with typed elements. Now, more tests pass.

### CODE REVIEW

File: test/timings/test_patterns.py [hunk 1]
```diff
@@ -336,3 +335,392 @@ def impl():
             _ = precedence_pattern.pattern_precedence
 
     benchmark.pedantic(impl, rounds=10, iterations=2)
+
+
+# ---------------------------------------------------------------------------
+# 7. get_match_candidates() vs get_match_candidates_count()
```
Comment (rocky) [inline test/timings/test_patterns.py:341]: 7. Smells like AI. <strike>Probably should remove the 7. Since I am not seeing 1. to 6.</strike>. I see them now.  Not complaining, but noting.

**Comment by rocky** [comment]
LGTM

File: test/timings/test_patterns.py (hunk 1 above)
Comment (mmatera) [reply test/timings/test_patterns.py:341]: Indeed, I am trying to use IA (Claude Sonnet) to detect what is going wrong with this module. Most of the tests and benchmarks it suggested sound reasonable to me, so I included them.
